rust/src/settings.rs

```diff
@@ -0,0 +1,184 @@
+    /// XAIN_PET__SUM=0.01
+    /// ```
+    pub sum: f64,
+
+    #[validate(range(min = 0, max = 1.0))]
+    /// The expected fraction of participants selected for submitting an updated local model for
+    /// aggregation. The value must be between `0` and `1`.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [pet]
+    /// update = 0.01
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN__PET__UPDATE=0.01
+    /// ```
+    pub update: f64,
+}
+
+#[derive(Debug, Validate, Deserialize)]
+/// REST API settings
+pub struct ApiSettings {
+    #[validate(url)]
+    /// The address to which the REST API should be bound.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [api]
+    /// bind_address = "localhost:8081"
+    /// # or
+    /// bind_address = "http://127.0.0.1:8081"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_API__BIND_ADDRESS=localhost:8081
+    /// ```
+    pub bind_address: String,
+}
+
+#[derive(Debug, Validate, Deserialize)]
+/// Mask settings
+pub struct MaskSettings {
+    /// The order of the finite group.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [mask]
+    /// group_type = "Integer"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_MASK__GROUND_TYPE=Integer
+    /// ```
+    pub group_type: GroupType,
+
+    /// The data type of the numbers to be masked.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [mask]
+    /// data_type = "F32"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_MASK__DATA_TYPE=F32
+    /// ```
+    pub data_type: DataType,
+
+    /// The bounds of the numbers to be masked.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [mask]
+    /// bound_type = "B0"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_MASK__BOUND_TYPE=B0
+    /// ```
+    pub bound_type: BoundType,
+
+    /// The bounds of the numbers to be masked.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [mask]
+    /// model_type = "M3"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAIN_MASK__MODEL_TYPE=M3
+    /// ```
+    pub model_type: ModelType,
+}
+
+impl Settings {
+    /// Loads and validates the coordinator settings via a configuration file.
+    /// Fails when the loading of the configuration file or its validation failed.
+    pub fn new(path: &str) -> Result<Self, SettingsError> {
+        let settings: Settings = Self::load(path)?;
+        settings.validate()?;
+        Ok(settings)
+    }
+
+    /// Loads the coordinator settings via a configuration file without validation.
+    /// Fails when the loading of configuration file failed.
+    pub fn new_dirty(path: &str) -> Result<Self, ConfigError> {
```

Comment:
in other parts of the code we use the `_unchecked()` suffix, but that usually implies a return value and not a return result/option. maybe `new_unvalidated()` is a more telling name?

Reply:
I changed it to `new_unchecked`. It sounds better 😁